homeassistant/components/netgear/sensor.py

```diff
@@ -37,23 +42,221 @@
         key="ssid",
         name="ssid",
         entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:wifi-marker",
     ),
     "conn_ap_mac": SensorEntityDescription(
         key="conn_ap_mac",
         name="access point mac",
         entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:router-network",
     ),
 }
 
 
+@dataclass
+class NetgearSensorEntityDescription(SensorEntityDescription):
+    """Class describing Netgear sensor entities."""
+
+    value: Callable = None
+    index: int = 0
+
+
+SENSOR_TRAFFIC_TYPES = [
+    NetgearSensorEntityDescription(
+        key="NewTodayUpload",
+        name="Upload today",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+        icon="mdi:upload",
+        value=lambda router, key: router.traffic_data.get(key)
```

Comment:
Pass the traffic data to the lambda instead since that's always what's needed. It will make the lambda shorter. We don't want lambdas spanning more then one line.

We also don't need to pass the key to the lambda since the description already knows its key. Just hardcode the key in the lambda, or alternatively just pass the value of the key directly to the lambda.

Reply:
thanks